vllm/model_executor/utils.py

```diff
@@ -83,3 +83,38 @@ def get_moe_expert_mapping(
             if child_map is not None:
                 return child_map()
         return []
+
+
+def disable_inductor_graph_partition(func):
+    """Decorator to disable inductor graph partition.
+    This is used to avoid nested cudagraph capture.
+
+    Example:
+    1. We apply torch.compile directly on some ops (e.g., grouped_topk) wrapped
+    in custom ops. Inductor graph partition applies cudagraph within the custom op.
+    2. At the same time, we compile the model which uses these custom ops. Inductor
+    graph partition also wraps each graph partition with CUDAGraph. Some partitions
+    may include custom ops, which has already been applied cudagraph. This leads to
+    nested cudagraph which is not supported.
+
+    This context manager should be wrapped around torch.compile calls within custom ops
+    to avoid the nested cudagraph capture.
+
+    Expected Usage:
+    @disable_inductor_graph_partition
+    @torch.compile()
+    def op_eager_code(...):
+        ...
+
+    Note that `@disable_inductor_graph_partition` should be applied before
+    `@torch.compile()`
```

Comment:
I think technically `torch.compile` is applied first:
```suggestion
    Note that `@disable_inductor_graph_partition` should be applied on top of 
    `torch.compile()`
```